static/app/views/performance/transactionSummary/transactionThresholdButton.tsx

```diff
@@ -0,0 +1,157 @@
+import {Client} from 'app/api';
+import Button from 'app/components/button';
+import {IconSettings} from 'app/icons';
+import {t} from 'app/locale';
+import {Organization, Project} from 'app/types';
+import {defined} from 'app/utils';
+import EventView from 'app/utils/discover/eventView';
+import withApi from 'app/utils/withApi';
+import withProjects from 'app/utils/withProjects';
+
+import TransactionThresholdModal, {
+  modalCss,
+  TransactionThresholdMetric,
+} from './transactionThresholdModal';
+
+type Props = {
+  api: Client;
+  organization: Organization;
+  transactionName: string;
+  projects: Project[];
+  eventView: EventView;
+  onChangeThreshold?: (threshold: number, metric: TransactionThresholdMetric) => void;
+};
+
+type State = {
+  transactionThreshold: number | undefined;
+  transactionThresholdMetric: TransactionThresholdMetric | undefined;
+  loadingThreshold: boolean;
+};
+
+class TransactionThresholdButton extends Component<Props, State> {
+  state: State = {
+    transactionThreshold: undefined,
+    transactionThresholdMetric: undefined,
+    loadingThreshold: false,
+  };
+
+  componentDidMount() {
+    this.fetchTransactionThreshold();
+  }
+
+  getProject() {
+    const {projects, eventView} = this.props;
+    if (!defined(eventView)) {
+      return undefined;
+    }
+
+    const projectId = String(eventView.project[0]);
+    const project = projects.find(proj => proj.id === projectId);
+
+    return project;
+  }
+
+  fetchTransactionThreshold = () => {
+    const {api, organization, transactionName} = this.props;
+
+    const project = this.getProject();
+    if (!defined(project)) {
+      return;
+    }
+    const transactionThresholdUrl = `/organizations/${organization.slug}/project-transaction-threshold-override/`;
+
+    this.setState({loadingThreshold: true});
+
+    api
+      .requestPromise(transactionThresholdUrl, {
+        method: 'GET',
+        includeAllArgs: true,
+        query: {
+          project: project.id,
+          transaction: transactionName,
+        },
+      })
+      .then(([data]) => {
+        this.setState({
+          loadingThreshold: false,
+          transactionThreshold: data.threshold,
+          transactionThresholdMetric: data.metric,
+        });
+      })
+      .catch(() => {
+        const projectThresholdUrl = `/projects/${organization.slug}/${project.slug}/transaction-threshold/configure/`;
+        this.props.api
+          .requestPromise(projectThresholdUrl, {
+            method: 'GET',
+            includeAllArgs: true,
+            query: {
+              project: project.id,
+            },
+          })
+          .then(([data]) => {
+            this.setState({
+              loadingThreshold: false,
+              transactionThreshold: data.threshold,
+              transactionThresholdMetric: data.metric,
+            });
+          })
+          .catch(err => {
+            this.setState({loadingThreshold: false});
```

Comment:
I think this may intended behaviour, but I will ask anyways lol. Do we still want to set `loadingThreshold` to `false` even if both threshold requests fail? This would allow the user to still click the button and they would probably be able to set their threshold if nothing else breaks.

Reply:
The pattern I see in our codebase is to set the loading state to false. And it makes sense that it's no longer loading now that the API request has failed. Although you do bring up a good point about allowing users to interact with the modal if the request fails. I *think* it should be okay because the API endpoint will only allow them to do things they are allowed to do. I'll think about it some more.

Reply:
I found a case where the get request could fail ahha- if the user has `project-transaction-threshold-override` flag on but `project-transaction-threshold` off and they don't have a transaction threshold configured and it tries to fall back to the project threshold. I have this draft [PR](https://github.com/getsentry/sentry/pull/27290) that would fix it, but it's kinda gross. I'm leaning towards a careful EA rollout of `project-transaction-threshold-override` and once we are at 100% just fully replacing `project-transaction-threshold` with `project-transaction-threshold-override` everywhere. @Zylphrex maybe you have some input on this as you do a second pass?

Reply:
If I'm understanding this correctly, this shouldn't an issue at all because the set of users with access to `project-transaction-threshold-override` should always be a subset of the users with access to `project-transaction-threshold`. And once this is in GA, this constraint will no longer be a concern.

If that's the case, I wouldn't worry too much about this, and just make sure the roll out is correct.

A scenario where this may be problematic is for on-prem where users can turn on whichever flag they want, but this is an edge case that I don't think we should think too hard about as the problem goes away once this is in GA.